webapp/src/main/java/info/rmapproject/webapp/service/DataDisplayServiceImpl.java

```diff
@@ -361,27 +374,34 @@ private ResourceDescription getObjectResourceDescription(String resource, List<R
 	 */
 	@Override
 	public ResultBatch<RMapTriple> getResourceBatch(String resourceUri, Integer offset, PaginatorType view) throws Exception {
-		URI uri = new URI(resourceUri);
-
-		RMapSearchParams params = paramsFactory.newInstance();
-		params.setStatusCode(RMapStatusFilter.ACTIVE);
-		params.setExcludeTypes(true);	
-		params.setOffset(offset);
-		if (view.equals(PaginatorType.RESOURCE_GRAPH)){
-			params.setExcludeLiterals(true);	
-			params.setLimit(maxResGraphRelationships);
-		} else {
-			params.setLimit(maxTableRows);
+		ResultBatch<RMapTriple> triplebatch = null;
+		try {
+			URI uri = new URI(resourceUri);
+	
+			RMapSearchParams params = paramsFactory.newInstance();
+			params.setStatusCode(RMapStatusFilter.ACTIVE);
+			params.setExcludeTypes(true);	
+			params.setOffset(offset);
+			if (view.equals(PaginatorType.RESOURCE_GRAPH)){
+				params.setExcludeLiterals(true);	
+				params.setLimit(maxResGraphRelationships);
+			} else {
+				params.setLimit(maxTableRows);
+			}
+			
+			triplebatch = rmapService.getResourceRelatedTriples(uri, params);
+	    	
+			//if there are no triples, don't load an empty screen, show a not found error.
+			//note that because only connected graphs are allowed, every URI should have at
+			//least one uri link.
+	    	if (triplebatch.size()==0)	{
+	    		throw new RMapObjectNotFoundException();
+	    	}
+		} catch (Exception ex) {
```

Comment:
If all you're doing here is re-throwing the exception, the catch block can be eliminated, because `Exception` is declared to be thrown in the method signature.  It just makes the code a little more readable.